Happy Dance/Assets/_Scripts/GameTimer.cs

```diff
@@ -0,0 +1,57 @@
+﻿using System.Collections;
+
+using UnityEngine;
+using UnityEngine.Assertions;
+using UnityEngine.Playables;
+
+using Liminal.Core.Fader;
+using Liminal.SDK.Core;
+
+/// <summary>
+/// GameTimer is sued to control the ending of the happy dance experience.
+/// </summary>
+public class GameTimer 
+    : MonoBehaviour
+{
+    public float SecondsLeft;
+    public float VolumeFadeSpeed;
+    public DanceDetector PlayerDanceDetector;
+    public PlayableDirector Director;
+
+    void Start()
+    {
+        Assert.IsNotNull(PlayerDanceDetector, "PlayerDanceDetector must not be null");
+        StartCoroutine(TimerCoro());
+    }
+
+	IEnumerator TimerCoro()
+	{
+        while (SecondsLeft > 0)
+        {
+            SecondsLeft -= Time.deltaTime;
+            yield return new WaitForEndOfFrame();
+        }
+
+        PlayerDanceDetector.Active = false;
+
+        yield return new WaitForSeconds(5f);
+        Director.gameObject.SetActive(true);
+
+        yield return new WaitForSeconds((float) Director.duration + 0.1f);
```

Comment:
(Not a must since it works, more of a do it if you want)

you can check against Director.state == PlayState.IsPlaying or if you're worried about it not yet been played, you can use Director.time in a while loop. 

One of useful things about a PlayableDirector is the ability to mess around with the timeline during playmode. Basing it on Director.duration can lead to undesired results for devs who don't actually know the experience well enough.